jerry-core/jerry.cpp

```diff
@@ -899,6 +900,65 @@ bool jerry_api_get_object_field_value (jerry_api_object_t *object_p,
 }
 
 /**
+ * Applies the given function to the every fields in the objects
+ *
+ */
+bool
+jerry_api_foreach_object_field (jerry_api_object_t *object_p, /**< object */
+                               jerry_object_field_foreach_t foreach_p, /**< foreach function */
+                               void *user_data) /**< user data for foreach function */
+{
+  jerry_assert_api_available ();
+
+  for (ecma_property_t *property_p = ecma_get_property_list (object_p);
+       property_p != NULL;
+       property_p = ECMA_GET_POINTER (ecma_property_t, property_p->next_property_p))
+  {
+    ecma_string_t *property_name_p;
+    if (property_p->type == ECMA_PROPERTY_NAMEDDATA)
+    {
+      property_name_p = ECMA_GET_NON_NULL_POINTER (ecma_string_t,
+                                                   property_p->u.named_data_property.name_p);
+    }
+    else if (property_p->type == ECMA_PROPERTY_NAMEDACCESSOR)
+    {
+      property_name_p = ECMA_GET_NON_NULL_POINTER (ecma_string_t,
+                                                   property_p->u.named_accessor_property.name_p);
+    }
+    else
+    {
+      continue;
+    }
+    if (!ecma_is_property_enumerable (property_p))
+    {
+      continue;
+    }
+
+    jerry_api_value_t field_value;
+    ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+    ECMA_TRY_CATCH (property_value, ecma_op_object_get (object_p, property_name_p),
```

Comment:
The ECMA_TRY_CATCH should be used the following way in this case:
```c
ECMA_TRY_CATCH(...)

jerry_api_convert...

if(!foreach_p(..))
{
  // note you should not break out from here, because the finalize would not be executed in that case.
   ...
}
jerry_api_release_value(...);

ECMA_FINALIZE(..)
```